addons/web_editor/static/src/js/backend/convert_inline.js

```diff
@@ -552,11 +552,14 @@ async function toInline($editable, cssRules, $iframe) {
     $editable.removeClass('odoo-editor-editable');
     const editable = $editable.get(0);
     const iframe = $iframe && $iframe.get(0);
+    const wysiwyg = $editable.data('wysiwyg');
     const doc = editable.ownerDocument;
-    cssRules = cssRules || doc._rulesCache;
+    cssRules = cssRules || wysiwyg && wysiwyg._rulesCache;
     if (!cssRules) {
         cssRules = getCSSRules(doc);
-        doc._rulesCache = cssRules;
+        if (wysiwyg) {
+            wysiwyg._rulesCache = cssRules;
+        }
```

Comment:
> Wouldn't the rules be different from one mailing template to another ?
Only the contents of the design style element (the one edited by the design tab).

> Moreover, the commit message does not give any actual reason why the current code is wrong except "it will complicate things down the line" which does nothing to convince the reader that the change was necessary.

It's because it's actually to make sure a wip branch of Nicolas doesn't break this. But now I remember the reason he needed this so I'll update :-D The reason is OWL destroys the DOM instead of hiding it when you for instance just switch tabs on the form view, so we lose the cache every time. So Nicolas' branch makes changes to account for that, and those changes would break my next commit so I anticipate it.

> If yes then wouldn't we run into trouble now that switching records does not destroy and recreate the widgets thanks to owl ?

We don't reinstanciate wysiwyg when switching records anymore? If so indeed that's a problem. But that just means we need to flush this cache when switching records. I guess that would be in `wysiwyg.resetEditor`.

Reply:
Sorry but I need to ask. You don't sound so sure in your reply but I see you did change the code. Did you test if my hypothesis was right ? Was there a bug in that case ? And did you test that resetting it in resetEditor solves that bug ? I suppose so but having the comment be unsure and the code already changed makes me uncomfortable about it... ^^'

Reply:
Sorry I wasn't clear, yes I tested and your hypothesis was right, as well as mine: when switching records, we don't instanciate a new wyswyg but we pass through resetEditor so the change fixes it.